commons/com.b2international.index/src/com/b2international/index/es/admin/EsIndexAdmin.java

```diff
@@ -224,17 +224,30 @@ public void create() {
 					SortedSet<String> incompatibleChanges = Sets.newTreeSet();
 					final JsonDiff schemaChanges = JsonDiff.diff(currentTypeMapping, newTypeMapping);
 					schemaChanges.forEach(change -> {
+						
 						if (change.isAdd()) {
+							
+							// XXX object type is the default type, so if the current mapping does not contain this node, we shouldn't trigger an update
+							if (change.getFieldPath().endsWith("type") && "object".equals(change.serializeValue())) {
+								return;
+							}
+							
 							compatibleChanges.add(change.getFieldPath());
+							
 						} else if (change.isMove() || change.isReplace()) {
 							incompatibleChanges.add(change.getFieldPath());
 						} else if (change.isRemove()) {
+							
 							// XXX while remove is bad it is hard to detect true incompatibility where we try to support dynamic fields (like Maps)
 							// raise the incompatibility warning only when the root field is being reported, not a nested property under the root property
-							if (!change.getFieldPath().contains("/properties")) {
-								incompatibleChanges.add(change.getFieldPath());
+							if (change.getFieldPath().contains("/properties")) {
+								return;
```

Comment:
Not sure about this part, the only this PR meant to change is the handling of the default type `object`. I would revert this back to the original state as changing the mapping of a nested object to something else is currently not possible.

Reply:
The inverted form doesn't rely on a sharp reader to spot the leading `!`:

```java
if (change.getFieldPath().contains("/properties")) {
  return;
}
							
incompatibleChanges.add(change.getFieldPath());
```